### PR TITLE
Fixes initial delay when starting Pods with AFS volumes

### PR DESCRIPTION
**Pull request by chriserik**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
When starting pods with AFS volumes, there is an initial delay of five seconds until the pod is transitioning from the "Scheduled" state. The reason for this is that currently the volume mounting happens inside a `wait.Poll` which will initially wait a specified interval (currently 5 seconds) before execution. 

Event log for comparison with and without AFS volume attached (seconds since event):
Screenshot 2020-06-09 at 23 40 27

This PR introduces an early return in case of a successful mount, which in the best case shortens Pod startup times by five seconds, speeding up scenarios where lots of Pods are scheduled with a high frequency (e.g. when using projects like https://argoproj.github.io/).

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by feiskyer** [comment]
@andyzhangx could you help to have a look?

File: pkg/volume/azure_file/azure_file.go [hunk 1]
```diff
@@ -302,6 +302,12 @@ func (b *azureFileMounter) SetUpAt(dir string, mounterArgs volume.MounterArgs) e
...
 	}
 
+	// Early exit if mount succeeded
+	err = b.mounter.MountSensitive(source, dir, "cifs", mountOptions, sensitiveMountOptions)
+	if err == nil {
+		return nil
+	}
+
 	mountComplete := false
 	err = wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
```
Comment (andyzhangx) [inline pkg/volume/azure_file/azure_file.go:312]: could you also change this to `1*time.Second`? thanks.

**Review by andyzhangx — COMMENTED** [review]
thanks for the finding! @chriserik 
could you also link this PR to https://github.com/kubernetes/kubernetes/issues/93025
set:
fixes #93025

Will cherry-pick to old release, thanks.

File: pkg/volume/azure_file/azure_file.go (hunk 1 above)
Comment (chriserik) [reply pkg/volume/azure_file/azure_file.go:312]: sure thing - you mean the poll interval, right?

**Comment by andyzhangx** [comment]
hi @chriserik I have worked out a new PR to address this issue: https://github.com/kubernetes/kubernetes/pull/93052, we should  use `wait.PollImmediate` to wait for mount time out, original `wait.Poll` is replaced by `wait.PollImmediate`. Let me know if you have any questions, thanks.

**Comment by chriserik** [comment]
Ok then I assume I can close this PR?

**Comment by andyzhangx** [comment]
> Ok then I assume I can close this PR?

yes, thanks for the finding!

**Pull request closed without merging**